relay-dynamic-config/src/global.rs

```diff
@@ -372,6 +373,39 @@ where
     }
 }
 
+fn deserialize_host<'de, D>(deserializer: D) -> Result<Vec<Host>, D::Error>
+where
+    D: serde::de::Deserializer<'de>,
+{
+    // Stolen from https://users.rust-lang.org/t/need-help-with-serde-deserialize-with/18374
+    struct HostVisitor;
+
+    impl<'de> de::Visitor<'de> for HostVisitor {
+        type Value = Vec<Host>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("a vector containing string of host data")
+        }
+
+        fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+        where
+            A: de::SeqAccess<'de>,
+        {
+            let mut values = Vec::<Host>::new();
+            while let Some(value) = seq.next_element()? {
+                let this = Host::parse(value);
+                if let Ok(t) = this {
+                    values.push(t)
+                }
+            }
```

Comment:
This visitor still fails if either one of the elements in the list has the wrong type or even the entire object is of the wrong type (more likely).

We can keep this fallible and not ignore invalid hosts (it's not user config, they should be always valid) and instead handle the entire error of the visitor at the `deserialize.deserialize_*` call.